.github/workflows/integration-tests.yml

```
@@ -0,0 +1,45 @@
+name: Collector Integration Tests
+
+on: 
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
+    branches:
+      - master
+  push:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  required-integration-tests:
+    uses: ./.github/workflows/integration-tests-vm-type.yml
+    strategy:
+      # ensure that if one part of the matrix fails, the
+      # rest will continue
+      fail-fast: false
+      matrix:
+        vm_type:
+          - rhel
+          - ubuntu-os
+    with:
+      vm_type: ${{ matrix.vm_type }}
+    secrets: inherit
+
+  all-integration-tests:
+    uses: ./.github/workflows/integration-tests-vm-type.yml
+    if: contains(github.event.pull_request.labels.*.name, 'all-integration-tests')
```

This we might want to run these on master/release too
```suggestion
    if: contains(github.event.pull_request.labels.*.name, 'all-integration-tests') || github.event_name == 'push'
```